### PR TITLE
Add support for suse 11 to service resource

### PR DESCRIPTION
**Pull request by spuranam**
This PR address [issue 627](https://github.com/chef/inspec/issues/627).

### CODE REVIEW

File: lib/resources/service.rb [hunk 1]
```diff
@@ -135,8 +135,15 @@ def select_service_mgmt # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticCom
         WindowsSrv.new(inspec)
       elsif %w{freebsd}.include?(family)
         BSDInit.new(inspec, service_ctl)
-      elsif %w{arch opensuse}.include?(family)
+      elsif %w{arch}.include?(family)
         Systemd.new(inspec, service_ctl)
+      elsif %w{suse opensuse}.include?(family)
+        version = inspec.os[:release].to_i
+        if (%w{ suse opensuse }.include?(family) && version >= 12)
```
Comment (chris-rock) [inline lib/resources/service.rb]: I think we can focus on version checking here. No need to test the family again

Comment (spuranam) [inline lib/resources/service.rb]: @chris-rock giving it another try, let me know.

File: lib/resources/service.rb [hunk 2]
```diff
@@ -135,8 +135,14 @@ def select_service_mgmt # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticCom
         WindowsSrv.new(inspec)
       elsif %w{freebsd}.include?(family)
         BSDInit.new(inspec, service_ctl)
-      elsif %w{arch opensuse}.include?(family)
+      elsif %w{arch}.include?(family)
         Systemd.new(inspec, service_ctl)
+      elsif %w{suse opensuse}.include?(family)
+        if (inspec.os[:release].to_i >= 12)
```
Comment (chris-rock) [inline lib/resources/service.rb]: please use `if inspec.os[:release].to_i >= 12` to meet our lint guidelines

**Comment by chris-rock** [comment]
That addition looks really good now.

Also please add `, Metrics/MethodLength` to https://github.com/chef/inspec/pull/638/files#diff-100aafe8eee954060cc58510cf55ba01R97 This ensures that the tests get green again.

**Comment by spuranam** [comment]
@chris-rock just checked the changes suggested above

**Comment by chris-rock** [comment]
Thanks @spuranam for this addition

**Pull request merged**